src/components/overlays/quiz-overlay.js

```diff
@@ -0,0 +1,379 @@
+import { PLAY_STATES } from '../../constants.js';
+import { IocRequesterMixin } from '../../mixins/ioc-requester.js';
+import { BindingHelpersMixin } from '../../mixins/binding-helpers.js';
+import { LocalizationMixin } from '../../mixins/localization.js';
+import '../../styling/overlay--style-module.js';
+import { PolymerElement, html } from '@polymer/polymer';
+
+class QuizOverlay extends BindingHelpersMixin(IocRequesterMixin(LocalizationMixin(PolymerElement))) {
+  static get template() {
+    return html`
+      <style type="text/css" include="overlay--style-module">
+        label {
+          font-weight: normal;
+        }
+
+        a[disabled] {
+          pointer-events: none;
+          color: grey;
+        }
+
+        h3 {
+          color: inherit;
+          font-size: 24px;
+        }
+
+        h5 {
+          color: inherit;
+          font-size: 14px;
+        }
+
+        #container__quiz-overlay {
+          display: flex;
+          align-items: center;
+          justify-content: center;
+          flex-direction: column;
+          background-color: rgba(0, 0, 0, 0.85);
+          color: white;
+        }
+
+        .container__centered-row {
+          display: flex;
+          align-items: center;
+          justify-content: center;
+          flex-direction: row;
+        }
+
+        .container__answer-box {
+          display: flex;
+          align-items: left;
+          justify-content: left;
+          flex-direction: column;
+        }
+
+        .button__quiz {
+          margin-top: 10px;
+          text-decoration: none;
+          color: inherit;
+          font-weight: bold;
+        }
+
+        .quiz__highlighting-none {
+          color: inherit;
+        }
+
+        .quiz__highlighting-correct {
+          color: lime;
+        }
+
+        .quiz__highlighting-wrong {
+          color: red;
+        }
+
+        .quiz__highlighting-strikethrough label {
+          text-decoration: line-through;
+        }
+
+        #input__freetext-answer {
+          background: transparent;
+          border: none;
+          border-bottom: 1px dashed;
+          text-align: center;
+          width: 300px;
+          color: inherit;
+        }
+      </style>
+
+      <div id="container__quiz-overlay" class="overlay" style$="visibility: [[ifThenElse(_isVisible, 'visible', 'hidden')]];">
+        <h3 id="text__quiz-question">[[_currentQuestion.text]]</h3>
+
+        <template is="dom-if" if="[[_isTextQuestion(_currentQuestion)]]">
+          <div class$="[[_getValidationClass(_currentQuestion, _correctAnswersShown, _correctAnswers, _isAnswerCorrect, null)]]">
```

Comment:
You mean that you don't like how they are colored? I don't have an opinion about that, we can change it, I'd just need to know how exactly to highlight them. I think we have about 3 choices:
- Highlight what the user entered (correctly selected and correctly not selected is green. currently implemented)
- Highlight the answers (correct answer is green, wrong answer is red, user selection does not matter)
- Some intermediate approach (if all selections are correct, make everything green. If some selection is wrong. mark that selection red and leave the rest white)

All off them seem misleading in some specific situations.

About the code that applies the styling: I don't really care, so I can change it so that the js sets the classes, but I will do that after we decide what styling exactly we want.

Reply:
No, sorry - I was referring to the code of the styling. The way questions are colored is fine with me :D

Reply:
Hmm. I just thought about it and I don't really think it's going to be much less code or much cleaner since we will still somehow need all the branching. I'll try implementing it and see how it turns out.